pkg/serviceaccounts/oauthclient/oauthclientregistry.go

```diff
@@ -389,15 +390,22 @@ func getScopeRestrictionsFor(namespace, name string) []oauthapi.ScopeRestriction
 
 // getServiceAccountTokens returns all ServiceAccountToken secrets for the given ServiceAccount
 func (a *saOAuthClientAdapter) getServiceAccountTokens(sa *kapi.ServiceAccount) ([]string, error) {
-	allSecrets, err := a.secretClient.Secrets(sa.Namespace).List(metainternal.ListOptions{})
+	allSecrets, err := a.secretClient.Secrets(sa.Namespace).List(metav1.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
+	sav1 := &kapiv1.ServiceAccount{}
+	kapiv1.Convert_api_ServiceAccount_To_v1_ServiceAccount(sa, sav1, nil)
 
 	tokens := []string{}
 	for i := range allSecrets.Items {
-		secret := allSecrets.Items[i]
-		if serviceaccount.IsServiceAccountToken(&secret, sa) {
+		secret := &allSecrets.Items[i]
+		secretv1 := &kapiv1.Secret{}
+		err := kapiv1.Convert_api_Secret_To_v1_Secret(secret, secretv1, nil)
+		if err != nil {
+			return nil, err
+		}
+		if serviceaccount.IsServiceAccountToken(secretv1, sav1) {
```

Comment:
no need to convert I can see... use `serviceaccount.InternalIsServiceAccountToken`?